packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -336,7 +337,7 @@ export class CdkToolkit {
 
       if (!stack.environment) {
         // eslint-disable-next-line max-len
-        throw new Error(
+        throw new AuthenticationError(
```

Comment:
I think the credentials part here is misleading. The error is that we cannot detect an environment. We try to use credentials in certain cases, but it's not the primary aspect.

```suggestion
        throw new ToolkitError(
```